### ISSUE TITLE
Formspec element textarea[] doesn't scale well vertically

### ISSUE THREAD

**Issue by ClobberXD** [minetest/minetest#8474]
##### Issue type

 - Bug report
 
##### Minetest version

```
5.1.0-dev-426bdba7
```

##### Summary

Formspec element `textarea[]` doesn't respect the `h` property, and falls short of the expected height.

##### Steps to reproduce

Create a mod and show the following formspec:
```
size[8,8]
textarea[0,0;4,8;textarea;Textarea;This should've been as tall as the formspec itself!]
label[4,8;This label is at y=8 for comparison]
```

**Comment by rubenwardy** [comment]
most formspec elements are like this, there's not much we can do without breaking compatibility

**Comment by SmallJoker** [comment]
There's likely no fix for this problem unless there's a way for mods to bump the formspec version individually.
If it helps, here's the code to calculate the correct coordinates:
https://github.com/minetest/minetest/blob/38b94f248ad2473b8c49e9832215071e8a79944f/builtin/mainmenu/dlg_settings_advanced.lua#L713-L727

**Comment by ClobberXD** [comment]
> There's likely no fix for this problem unless there's a way for mods to bump the formspec version individually.

That's sad... :/

> If it helps, here's the code to calculate the correct coordinates: ...

I'd have to resort to eye-balling the approximate height to fit the expected size, otherwise. So this is indeed quite helpful, thanks. :)

**Comment by Linuxdirk** [comment]
> most formspec elements are like this,

Actually there are no formspec elements being consistent to even one of the other formspec elements.

![Nvawt8i](https://user-images.githubusercontent.com/46120351/56915595-3e66e180-6ab7-11e9-8fb4-6472b2585a04.gif)


All of the elements are set to position 0,0 and size 2,2.

**Comment by rubenwardy** [comment]
that's actually a very helpful gif

**Comment by Linuxdirk** [comment]
> that's actually a very helpful gif

Nah, not really. It just demonstrate how formspec elements elements reacto to position 0,0 and size 2,2.

In the forums I posted the mod used for creation of this. Once the forums are accessible again you can get the source there since I do not have the mod anymore because I only used it to create the screenshots for this animation.

**Comment by sorcerykid** [comment]
Actually this is due to a typo in the dimension calculations. I doubt it was intentional, despite how wacky the coordinate system is. I encountered this while developing my Scarlet mod.

_In the pseudocode below, "slot_size" refers to the size of an inventory slot (in code: imgsize) and "cell_margin" refers to the size of just the gutter (in code: spacing - imgsize), and "cell_size" refers to the size of an inventory slot in addition to the gutter (in code: spacing)._

Here is the erroneous code in question:
https://github.com/minetest/minetest/blob/429a98964859b83016f2eb47a47a08ab8dc3c57e/src/gui/guiFormSpecMenu.cpp#L1153

Currently, the y dimension for textarea[] elements is (mis)calculated as

`dim_y = stof( val ) * slot_size_y - cell_margin_y`

However, this makes no sense. Multiplying by slot_size will get a multiple of inventory slot sizes, but it will not be aligned to the inventory grid (except if val is 1), so what is the purpose of subtracting cell_margin? Clearly this was a mistake and the intended formula is

`dim_y = stof( val ) * cell_size_y - cell_margin_y`

Virtually all other element dimensions are calculated this way. So I suspect that sapier made a typographical error and it slipped by during testing.

Similarly, the horizontal dimension of dropdown[] elements is miscalculated.

`dim_x = stof( val ) * cell_size_y`

Here is the erroneous code in question:
https://github.com/minetest/minetest/blob/429a98964859b83016f2eb47a47a08ab8dc3c57e/src/gui/guiFormSpecMenu.cpp#L893

It certainly appears that sapier copied and pasted this code from elsewhere and forgot to correct the axis. Clearly this should be

`dim_x = stof( val ) * cel_size_x`

I think both of these can be rightfully qualified as bugs and there seems justification for them to be corrected probably in conjunction with PR #8524 .

**Comment by SmallJoker** [comment]
Fixed by `real_coordinates`
